packages/perseus-core/src/analytics.ts

```diff
@@ -0,0 +1,26 @@
+import type {AnalyticsEvent} from "@khanacademy/event-schemas";
+
+// Magic based on:
+//   * https://echobind.com/post/slicing-typescript-literal-strings
+//   * https://www.typescriptlang.org/docs/handbook/utility-types.html
+//
+// This type removes keys that Perseus cannot fill in. It is expected that the
+// hosting application will populate these missing values. We represent this as
+// a type so taht the host can easily detect which fields in the type are
+// missing at "compile time".
+type RemoveUnsupportedKeys<T> = {
+    [P in keyof T as P extends `contentPath_${string}`
+        ? never
+        : P extends `currentPageURL` // STOPSHIP: To be removed from schema!
+        ? never
+        : P]: T[P];
+};
```

Comment:
This is a really complicated type.  I would try to extract a helper function for the key part of the mapped type and create an analytics.typestest.ts file that includes some type tests that verifies that `RemoveUnsupportedKeys` has the desired behavior.

Reply:
Great idea!!   Since you read this PR I was able to simplify this some (we no longer need to remove `currentPageURL`), but I love the idea of some tests.